plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,697 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022, Castor Sky (@castorsky) <csky57@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_disk
+short_description: Management of a disk of a Qemu(KVM) VM in a Proxmox VE cluster.
+version_added: 5.5.0
+description:
+  - Allows you to perform some supported operations on a disk in Qemu(KVM) Virtual Machines in a Proxmox VE cluster.
+author: "Castor Sky (@castorsky) <csky57@gmail.com>"
+options:
+  name:
+    description:
+      - The (unique) name of the VM.
+      - Required only one of I(name) or I(vmid).
```

Comment:
The module specification is:
```
        required_together=[('api_token_id', 'api_token_secret')],
        required_one_of=[('name', 'vmid'), ('api_password', 'api_token_id')],
        required_if=[
            ('state', 'present', ('storage', 'size')),
            ('state', 'grown', ['size'])
        ],
        required_by={'target_disk': 'target_vmid'},
        supports_check_mode=False,
        mutually_exclusive=[('target_vmid', 'target_storage')]
```

Which does not correspond to the documentation. The `required_one_of` argument for `AnsibleModule` indicates that, out of that list (in this case the tuple `('name', 'vmid')`), **at least one of them must be passed**, but it does not prevent the user from passing them both.

If you want to enforce that behaviour (only one of them can be passed), then you must also add that tuple to the `mutually_exclusive` argument for `AnsibleModule`.

See https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#dependencies-between-module-options for reference on these